### PR TITLE
Modify diskann memory estimation method

### PR DESCRIPTION
**Pull request by cqy123456**

### CODE REVIEW

**Comment by cqy123456** [comment]
run-cpu-e2e

**Comment by cqy123456** [comment]
run-cpu-e2e

File: internal/querynode/segment_loader.go [hunk 1]
```diff
@@ -700,7 +718,21 @@ func GetStorageSizeByIndexInfo(indexInfo *querypb.FieldIndexInfo) (uint64, uint6
...
+		searchCacheProportion, err := strconv.ParseFloat(indexParams[indexparams.SearchCacheBudgetKey], 64)
+		if err != nil {
+			return 0, 0, fmt.Errorf("%s not exist in index params", indexparams.SearchCacheBudgetKey)
+		}
+		dim, err := strconv.ParseInt(indexParams["dim"], 10, 64)
+		if err != nil {
+			return 0, 0, fmt.Errorf("dim not exist in index params")
+		}
+		rawDataSize := indexparams.GetRowDataSizeOfFloatVector(indexInfo.NumRows, dim)
+		neededMemSize := float64(rawDataSize)*(PQCodeProportion) + searchCacheProportion*DiskANNCacheExpansionFactor*(1024*1024*1024)
```
Comment (xige-16) [inline internal/querynode/segment_loader.go:735]: add a comment would be better

File: internal/util/indexparams/disk_index_params.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ const (
 	MaxBeamWidth  = 16
 )
 
-func getRowDataSizeOfFloatVector(numRows int64, dim int64) int64 {
```
Comment (liliu-z) [inline internal/util/indexparams/disk_index_params.go:48]: Why this change

File: internal/querynode/segment_loader.go [hunk 3]
```diff
@@ -752,8 +784,14 @@ func (loader *segmentLoader) checkSegmentSize(collectionID UniqueID, segmentLoad
 						zap.Int64("indexBuildID", fieldIndexInfo.BuildID))
 					return err
 				}
-				usedMemAfterLoad += neededMemSize
 				usedLocalSizeAfterLoad += neededDiskSize
+				// diskann not need to copy data
+				indexType, _ := funcutil.GetAttrByKeyFromRepeatedKV("index_type", fieldIndexInfo.IndexParams)
+				if indexType == indexparamcheck.IndexDISKANN {
+					usedMemAfterLoad += uint64(float64(neededMemSize) / Params.QueryNodeCfg.LoadMemoryUsageFactor)
```
[the remark this answers is not in the document]
Comment (liliu-z) [reply internal/querynode/segment_loader.go:839]: I think this is a bug
I guess we thought `memLoadingUsage = usedMemAfterLoad / LoadMemoryUsageFactor`, and for DiskANN we don't need the loading expansion factors. So we div LoadMemoryUsageFactor as a preprocessing.
But `memLoadingUsage = usedMemAfterLoad + maxSegmentSize * concurrency * LoadMemoryUsageFactor`, so we should preprocess the second part instead of the first.
